File: my_project_typescript/easy_problems_typescript/from151to200/sum_two.ts
function twoSum(nums: number[], target: number): number[] {

    let hashtable = {}
    let lenNums: number = nums.length

    for (let i = 0; i < lenNums; i++){

        if (nums[i] in hashtable){
            return [i, hashtable[nums[i]]]
        }
        hashtable[target-nums[i]] = i
    }

    return [-1,-1]



};

File: src/my_project/interviews/interview_amazon/round_2/sum_two.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        dic_answer = dict()

        for k, v in enumerate(nums):

            if v in dic_answer:
                return [dic_answer[v], k]
            else: 
                dic_answer[target - v] = k